File: FoX/iFaX/iFaX.3/exercise_3/xpather_ns.py
#!/usr/bin/python
import lxml.etree

namespaces = {'q':'http://www.example.com/quotes', 'f':'http://www.example.com/films'}

docRoot = lxml.etree.parse(source="monty_ns.xml")

answer = docRoot.xpath("/f:film/@name", namespaces)

print answer

File: FoX/iFaX/iFaX.3/exercise_3/xpather_ns.pl
#!/usr/bin/perl 
use XML::LibXML;

my $docRoot =XML::LibXML->new()->parse_file('monty_ns.xml');

my $context = XML::LibXML::XPathContext->new($docRoot);
$context->registerNs('q', 'http://www.example.com/quotes');
$context->registerNs('f', 'http://www.example.com/films');

my $result = $context->findnodes('/f:film/@name');

print $result . "\n";


